test/functional/ex_cmds/mksession_spec.lua

```diff
@@ -0,0 +1,50 @@
+-- Test generation of session files
+
+local helpers = require('test.functional.helpers')(after_each)
+
+
+describe('A session file', function()
+  -- The file name use for each session
+  local session_file = 'Xtest-functional-ex_cmds-mksession.vim'
+
+  before_each(function()
+    helpers.clear()
+  end)
+
+  -- Each test should start with a clean session file
+  after_each(function()
+    os.remove(session_file)
+  end)
+
+  it('restores tab-local working directories', function()
+    local tab_dir = 'Xtest-functional-ex_cmds-mksession_spec.tab'
+    local cwd_dir = helpers.call('getcwd')
+    lfs.mkdir(tab_dir)
+
+    -- There need to be some files being edited, so we will use /dev/null
+    helpers.command('edit /dev/null')
+    helpers.command('tabnew')
+    helpers.command('edit /dev/null')
+    helpers.command('tcd ' .. tab_dir)
+    helpers.command('tabfirst')
+    helpers.command('mksession! ' .. session_file)
+    helpers.command('qall!')
+
+    -- Create a new test instance of Nvim, and henceforth all test
+    -- utilities operate on this new instance.
+    local nvim2 = helpers.spawn(helpers.nvim_argv)
+    helpers.set_session(nvim2)
+    local pathsep = helpers.get_pathsep()  -- Needed for portable path strings
```

Comment:
This is used in one line only. Would be logical to just call `get_pathsep()` directly there.